<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <!-- Domyślny rozmiar okna -->
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>720</height>
   </rect>
  </property>
  
  <!-- Minimalny rozmiar 640×480 -->
  <property name="minimumSize">
   <size>
    <width>640</width>
    <height>480</height>
   </size>
  </property>

  <property name="windowTitle">
   <string>Micromouse</string>
  </property>

  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <!-- LEWA RAMKA (QGraphicsView) -->
    <item>
     <widget class="QFrame" name="leftframe">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayoutLeft">
       <item>
        <widget class="QGraphicsView" name="graphicsView"/>
       </item>
      </layout>
     </widget>
    </item>

    <!-- PRAWA RAMKA (panel sterowania) -->
    <item>
     <widget class="QFrame" name="rightframe">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayoutRight">
       <!-- Tytuł "Micromouse" -->
       <item>
        <widget class="QLabel" name="titleLabel">
         <property name="font">
          <font>
           <pointsize>72</pointsize>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Micromouse</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>

       <!-- Etykieta "Map Size" -->
       <item>
        <widget class="QLabel" name="mapSizeLabel">
         <property name="font">
          <font>
           <pointsize>20</pointsize>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Map Size</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>

       <!-- Etykieta z wartością suwaka (np. "10x10") -->
       <item>
        <widget class="QLabel" name="mapSizeValueLabel">
         <property name="font">
          <font>
           <pointsize>16</pointsize>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>10x10</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>

       <!-- Slider mapSlider (8..20) -->
       <item>
        <widget class="QSlider" name="mapSlider">
         <property name="minimum">
          <number>8</number>
         </property>
         <property name="maximum">
          <number>20</number>
         </property>
         <property name="value">
          <number>10</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksBothSides</enum>
         </property>
         <property name="tickInterval">
          <number>1</number>
         </property>
        </widget>
       </item>

       <!-- Przyciski Start / Stop / Restart -->
       <item>
        <layout class="QHBoxLayout" name="horizontalLayoutButtons">
         <item>
          <widget class="QPushButton" name="startButton">
           <property name="text">
            <string>Start</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="stopButton">
           <property name="text">
            <string>Stop</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="restartButton">
           <property name="text">
            <string>Restart</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>

       <!-- Etykieta "Speed" -->
       <item>
        <widget class="QLabel" name="speedTitleLabel">
         <property name="font">
          <font>
           <pointsize>16</pointsize>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Speed</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>

       <!-- QDial (1..5) -->
       <item>
        <widget class="QDial" name="speedDial">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>5</number>
         </property>
         <property name="value">
          <number>1</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>

       <!-- Etykieta pokazująca prędkość (np. "1x") -->
       <item>
        <widget class="QLabel" name="speedValueLabel">
         <property name="font">
          <font>
           <pointsize>16</pointsize>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>1x</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>

       <!-- Etykieta z animowanym GIF-em (spinner), domyślnie niewidoczna -->
       <item>
        <widget class="QLabel" name="loadingLabel">
         <property name="visible">
          <bool>false</bool>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>

       <!-- Etykieta "Results" (duży font), niewidoczna na start -->
       <item>
        <widget class="QLabel" name="resultsTitleLabel">
         <property name="font">
          <font>
           <pointsize>72</pointsize>
           <bold>true</bold>
          </font>
         </property>
         <property name="visible">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Results</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>

       <!-- Etykieta czasu - niewidoczna na start -->
       <item>
        <widget class="QLabel" name="timeLabel">
         <property name="visible">
          <bool>false</bool>
         </property>
         <property name="font">
          <font>
           <pointsize>16</pointsize>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>Time: 0 s</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>

       <!-- Etykieta liczby ruchów - niewidoczna na start -->
       <item>
        <widget class="QLabel" name="movesLabel">
         <property name="visible">
          <bool>false</bool>
         </property>
         <property name="font">
          <font>
           <pointsize>16</pointsize>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>Moves: 0</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>

      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
